test/AutoRest.TestServerLowLevel.Tests/dpg_update1.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using AutoRest.TestServer.Tests.Infrastructure;
+using dpg_update1_LowLevel;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class DpgUpdate1Test : TestServerLowLevelTestBase
+    {
+        [Test]
+        public Task DpgAddOptionalInput_NoParams_Test() => Test(async (host) =>
+        {
+            // compatible with dpg_initial
+            var result = await new ParamsClient(Key, host).HeadNoParamsAsync();
+            Assert.IsEmpty(result.Content.ToArray());
+            Assert.AreEqual(200, result.Status);
+            Assert.AreEqual(123, result.Headers.ContentLength);
+            var result2 = await new ParamsClient(Key, host).HeadNoParamsAsync(new Azure.RequestContext());
+            Assert.IsEmpty(result2.Content.ToArray());
+            Assert.AreEqual(200, result2.Status);
+            Assert.AreEqual(123, result2.Headers.ContentLength);
+            // use methods in dpg_update1
+            var result3 = await new ParamsClient(Key, host).HeadNoParamsAsync("newParam");
+            Assert.IsEmpty(result3.Content.ToArray());
+            Assert.AreEqual(200, result3.Status);
+            Assert.AreEqual(123, result3.Headers.ContentLength);
+            var result4 = await new ParamsClient(Key, host).HeadNoParamsAsync("newParam", new Azure.RequestContext());
+            Assert.IsEmpty(result4.Content.ToArray());
+            Assert.AreEqual(200, result4.Status);
+            Assert.AreEqual(123, result4.Headers.ContentLength);
+        });
+
+        [Test]
+        public Task DpgAddOptionalInput() => Test(async (host) =>
+        {
+            // compatible with dpg_initial
+            var result = await new ParamsClient(Key, host).GetRequiredAsync("param");
```

Comment:
Is there a way to reuse the code from the dpg_initial to ensure that it is identical?  This feels a little error-prone if someone modified one test file and not the other.  It may not be a big deal to do it this way for tests, though.

Reply:
Refactored to share dpg_initial test code.